cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/Artifacts.java

```diff
@@ -45,7 +45,14 @@ public static String getFileName(ArtifactId artifactId) {
   public static Type getConfigType(Class<? extends Application> appClass) {
     TypeToken<?> configType = TypeToken.of(appClass).resolveType(Application.class.getTypeParameters()[0]);
     if (Reflections.isResolved(configType.getType())) {
-      return configType.getType();
+      // Default the type to Config.class if the resolved type is not subclass of Config.
+      // It normally won't happen, unless someone generate the bytecode directly.
+      // This is for Scala. If the user don't specify any type parameter, Scala will automatically insert
+      // "scala.runtime.Nothing$" as the type parameter.
+      if (Config.class.isAssignableFrom(configType.getRawType())) {
```

Comment:
Why? What if something like clojure or groovy is doing that as well? It can only happen if someone write out the bytecode directly, I think it's save to say if it is not a `Config`, we give you a `Config` as it is enforced by the Java interface.